core-bundle/src/Filesystem/FilesystemItemIterator.php

```diff
@@ -97,6 +97,27 @@ public function all(callable $condition): bool
         return true;
     }
 
+    public function limit(int $numberOfElements): self
+    {
+        if ($numberOfElements < 0) {
```

Comment:
```suggestion
        if ($numberOfElements <= 0) {
```

Or not? Or maybe better just throw an `\OutOfBoundsException`? In which case would passing 0 or negative ever make any sense?

Reply:
IMHO providing 0 is a valid scenario. Think about yielding from multiple generators for instance. 

I'm fine with an exception for the `< 0` case - this probably makes for a better API. :+1: 
Is `\OutOfBoundsException` the right one, though, because the threshold is never dynamic? `\OutOfRangeException`?

Reply:
Yeah you're right :)

Reply:
IMO `\InvalidArgumentException` if `< 0`

Reply:
see e92f28bd18052740e57deb92dee213710930d1ba